### DISCUSSION TITLE
reTerminal DM RS485 Port with Node-RED | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#573] [Q&A] [answered]
# reTerminal DM RS485 Port with Node-RED | Seeed Studio Wiki

reTerminal DM RS485 Port with Node-RED

https://wiki.seeedstudio.com/reTerminal-DM-Node-Red-RS485/

**Comment by djurcola** [comment]
Hello, the RS485 (/dev/ttyACM1) is not available on both my ReTerminal units: 
Command: ls | grep "ttyA"
Response: ttyAMA0
Any ideas how to make serial available?

**Reply by xzxcessarr** [reply]
Hello, we need more information to accquire the error occur in your device:
1. use "dmesg | grep tty" and post the output result, please check your RS485 device by the USBto485 port and connect it well
2. check whether you have activate the serial interface in the system: use the "sudo raspi-config" check "the serial interface is enabled", and tell us whether your "the serial login shell" is disabled or not

**Reply by iGridAU** [reply]
Hello, thanks for your support - please see output below: 

```
dmesg | grep tty
[    0.000000] Kernel command line: coherent_pool=1M 8250.nr_uarts=1 snd_bcm2835.enable_headphones=0 snd_bcm2835.enable_hdmi=1 snd_bcm2835.enable_hdmi=0 snd_bcm2835.enable_hdmi=0  smsc95xx.macaddr=E4:5F:01:F3:D9:E6 vc_mem.mem_base=0x3ec00000 vc_mem.mem_size=0x40000000  console=tty1 root=PARTUUID=dcdfde03-02 rootfstype=ext4 fsck.repair=yes rootwait logo.nologo vt.global_cursor_default=0 console=tty3 loglevel=0
[    0.000372] printk: console [tty1] enabled
[    1.844180] fe201000.serial: ttyAMA0 at MMIO 0xfe201000 (irq = 35, base_baud = 0) is a PL011 rev2
[    1.853187] fe215040.serial: ttyS0 at MMIO 0xfe215040 (irq = 36, base_baud = 62500000) is a 16550
[    3.318228] systemd[1]: Created slice system-getty.slice.
[    6.352695] usb_ch343 1-1.3:1.0: ttyCH343USB0: usb to uart device
[    6.372005] usb_ch343 1-1.3:1.2: ttyCH343USB1: usb to uart device
```

For serial enable I have tried both of these options:

```
The serial login shell is disabled
The serial interface is enabled
```

and 

```
The serial login shell is enabled
The serial interface is enabled
```

I can still not see ttyACM1

**Reply by peterpanstechland** [reply, marked as answer]
Hi, Sorry for the inconvenience, at the time when I prepared this wiki the RS485 serial port was called ```/dev/ttyACM1```, since the new driver relocate the ```/dev/ttyACM1``` to ```/dev/ttyCH343USB1```, however this port name can not be detected by the "magnification glass" icon, which mean you need to type-in the exact port name as ```/dev/ttyCH343USB1```, BTW I will update this wiki in the mean time.

**Reply by djurcola** [reply]
Hi, thanks for your feedback. I have got the serial port working in NodeRed with the help of your comment and some additional changes, captured below:
Add dialup group to the NodeRed docker compose and make container privileged:
```
    privileged: true
    group_add:
      - 'dialout'
```

**Comment by j-broome** [comment]
Using latest version of bookworm and see /dev/ttyCH343USB1, but don't seem to be able to use it. Not using docker.

**Reply by nfs0619** [reply]
Thank you for your feedback. We have received your message and are currently arranging for debugging. If you do not receive further updates, please feel free to reach out to our technical support at [techsupport@seeed.io](mailto:techsupport@seeed.io) for further assistance. Thank you for your support and understanding.
